spec/support/shared_examples/can_have_registration_attributes.rb

```diff
@@ -402,4 +402,88 @@
       end
     end
   end
+
+  describe "#legal_entity_name" do
+    let(:person_a) { build(:key_person, :main, first_name: Faker::Name.first_name, last_name: Faker::Name.last_name) }
+    let(:key_people) { [person_a] }
+    let(:registered_company_name) { nil }
+    let(:tier) { WasteCarriersEngine::Registration::UPPER_TIER }
+    let(:company_name) { nil }
+    let(:resource) do
+      build(factory,
+            business_type: business_type,
+            tier: tier,
+            registered_company_name: registered_company_name,
+            key_people: key_people)
+    end
+
+    subject { resource.legal_entity_name }
+
+    shared_examples "returns registered name" do
```

Comment:
Can we clarify that this refers to the `registered_company_name` and that business name refers to the `company_name`